src/plugins/Workers/SyncWorker/utils/TransactionOrderer/methods/insertAtPos.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Add a transaction at a specific position
+ * @param {Transaction} transaction
+ * @param {number|null} pos
+ */
+module.exports = function insertAtPos(transaction, pos) {
+  const { transactionIds, transactions } = this;
+  if (pos != null) {
+    // eslint-disable-next-line no-param-reassign
+    if (pos > transactions.length) pos = transactions.length;
```

Comment:
It's kinda magic. You pass the wrong position but it works. Probably, it's better to throw an error?

Reply:
It is actually better, it shouldn't happen that you insert that far and if you did it's problematic, an error will at least told you about it.